src/lib/db.ts

```diff
@@ -277,72 +277,93 @@ export class DB {
 
     // problem document
     if (documentType === ProblemDocument) {
-      const problemDocument = documents.getProblemDocument(user.id);
-      if (problemDocument) return problemDocument;
-
-      const problemDocumentsRef = this.firebase.ref(this.firebase.getProblemDocumentsPath(user));
-      const problemDocumentsSnapshot = await problemDocumentsRef.once("value");
-      const problemDocuments: DBOfferingUserProblemDocumentMap = problemDocumentsSnapshot?.val();
-      const lastProblemDocument = findLast(problemDocuments, () => true);
-      return lastProblemDocument?.documentKey
-              ? this.openProblemOrPlanningDocument(ProblemDocument, lastProblemDocument.documentKey)
-              : this.createProblemOrPlanningDocument(ProblemDocument, defaultContent);
+      const requiredProblemDocument = documents.requiredDocuments[ProblemDocument];
+      if (requiredProblemDocument) {
+        const problemDocument = await requiredProblemDocument.promise;
+        if (problemDocument) return problemDocument;
+
+        const problemDocumentsRef = this.firebase.ref(this.firebase.getProblemDocumentsPath(user));
+        const problemDocumentsSnapshot = await problemDocumentsRef.once("value");
+        const problemDocuments: DBOfferingUserProblemDocumentMap = problemDocumentsSnapshot?.val();
+        const lastProblemDocument = findLast(problemDocuments, () => true);
+        return lastProblemDocument?.documentKey
+                ? this.openProblemOrPlanningDocument(ProblemDocument, lastProblemDocument.documentKey)
+                : this.createProblemOrPlanningDocument(ProblemDocument, defaultContent);
```

Comment:
Given that we only get here when there is no existing problem document, this code for finding the most recent of all the existing problem documents seems unnecessarily complicated. Presumably, we should always be going through the `createProblemOrPlanningDocument` path.